src/main/java/uk/gov/companieshouse/filinghistory/consumer/delta/InternalFilingHistoryApiMapper.java

```diff
@@ -0,0 +1,82 @@
+package uk.gov.companieshouse.filinghistory.consumer.delta;
+
+import static uk.gov.companieshouse.filinghistory.consumer.delta.MappingUtils.getValueFromField;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.filinghistory.ExternalData;
+import uk.gov.companieshouse.api.filinghistory.ExternalData.CategoryEnum;
+import uk.gov.companieshouse.api.filinghistory.ExternalData.SubcategoryEnum;
+import uk.gov.companieshouse.api.filinghistory.InternalData;
+import uk.gov.companieshouse.api.filinghistory.InternalFilingHistoryApi;
+
+@Component
+public class InternalFilingHistoryApiMapper {
+
+    private static final Pattern BARCODE_REGEX = Pattern.compile("^X");
+    private static final Pattern DOCUMENT_ID_REGEX = Pattern.compile("^...X", Pattern.CASE_INSENSITIVE);
+
+    private final DescriptionValuesMapper descriptionValuesMapper;
+    private final LinksMapper linksMapper;
+    private final OriginalValuesMapper originalValuesMapper;
+
+    public InternalFilingHistoryApiMapper(DescriptionValuesMapper descriptionValuesMapper, LinksMapper linksMapper, OriginalValuesMapper originalValuesMapper) {
+        this.descriptionValuesMapper = descriptionValuesMapper;
+        this.linksMapper = linksMapper;
+        this.originalValuesMapper = originalValuesMapper;
+    }
+
+    public InternalFilingHistoryApi mapJsonNodeToInternalFilingHistoryApi(final JsonNode topLevelNode, final TransactionKindResult kindResult, final String deltaAt, final String updatedBy) {
+        final JsonNode dataNode = topLevelNode.get("data");
+        final JsonNode originalValuesNode = topLevelNode.get("original_values");
+        final JsonNode descriptionValuesNode = dataNode.get("description_values");
```

Comment:
What happens when either of the above "values" fields are null? It should result in the field not being set on the request. Currently it might throw a null pointer or set an empty object.